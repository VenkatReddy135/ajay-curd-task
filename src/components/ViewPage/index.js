import React, { useState, useEffect } from "react";
import { useParams, useNavigate } from "react-router-dom";
import CustomForm from "../../shared_components/CustomForm";
import { getOne } from "../../utils/helperFunction";

const ViewEmployeeDetails = () => {
  const { id } = useParams();
  const navigate = useNavigate();
  const [viewData, setViewData] = useState({});
  const resource = `http://dummy.restapiexample.com/api/v1/employee`;

  useEffect(() => {
    const employeeId = id;
    const getOneData = async () => {
      const res = await getOne(resource, employeeId);
      const apiData = res.data;
      console.log(apiData.data);
      setViewData(apiData.data);
    };
    getOneData();
  }, [id, resource]);

  const backAction = () => {
    navigate(`/`);
  };

  return (
    <CustomForm
      formData={viewData}
      mode="view"
      buttonText="Back"
      buttonAction={backAction}
    />
  );
};

export default React.memo(ViewEmployeeDetails);
