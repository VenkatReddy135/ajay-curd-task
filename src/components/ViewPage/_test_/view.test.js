/* eslint-disable testing-library/no-node-access */
import React from "react";
import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ViewPage from "../index";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  useParams: () => {
    return {
      match: {
        id: "1",
      },
    };
  },
  useNavigate: jest.fn(),
}));

afterEach(cleanup);

describe("View Component Testing", () => {
  const renderedData = renderer.create(<ViewPage />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ViewPage />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });
});
