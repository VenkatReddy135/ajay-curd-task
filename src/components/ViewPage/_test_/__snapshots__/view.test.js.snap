// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Component Testing renders a snapshot 1`] = `
<div>
  <form
    className="CustomForm"
  >
    <div
      className="container border border-info p-4 mt-4"
    >
      <h2>
         Details:
      </h2>
      <div
        className="row"
      >
        <div
          className="col-4"
        >
          <label>
            Employee Name:
          </label>
          <input
            className="border-0 form-control inputWidth"
            placeholder=""
            type="text"
          />
        </div>
        <div
          className="col-4"
        >
          <label>
            Employee Age:
          </label>
          <input
            className="border-0 form-control inputWidth"
            placeholder=""
            type="text"
          />
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <label>
              Employee Salary:
            </label>
            <input
              className="border-0 form-control inputWidth"
              placeholder=""
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <button
            className="button btn btn-secondary backButton"
            onClick={[Function]}
          >
            Back
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
